[DOC-861][2024.2.5] Update SCRAM support in conn mgr. Add channel binding flag info.

vdamle-yb opened the pull request:
This doc diff updates the status of SCRAM auth support in connection manager and adds information about enabling channel binding for SCRAM (aka SCRAM-SHA-256-PLUS, disabled by default). These feature changes are present in 2024.2.5 and 2025.1.1. 
DOC-861
DOC-800

vdamle-yb committed:
Add information about SCRAM-SHA-256-PLUS gflag to TLS-auth doc page (preview,stable,2024.2)

vdamle-yb committed:
Update SCRAM support info for conn mgr (preview,stable,2024.2)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/yugabyte/yugabyte-db?pullRequest=28213) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/yugabyte/yugabyte-db?pullRequest=28213) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/yugabyte/yugabyte-db?pullRequest=28213) it.</sub>

vdamle-yb committed:
Fix formatting error

ddhodge committed:
Merge branch 'master' into pr/vdamle-yb/28213

ddhodge committed:
fix typo

ddhodge commented on docs/content/preview/secure/tls-encryption/tls-authentication.md:
````diff
@@ -152,6 +152,28 @@ Type "help" for help.
+
+To create a database, execute the following command:
+```sh
+$ ./bin/yb-ctl destroy && ./bin/yb-ctl create \
````
yb-ctl is practically deprecated. Is it possible to redo these examples using yugabyted

vdamle-yb replied:
Resolved. I had run this with `yugabyted` locally, but thought docs used `yb-ctl` or something like that. Replaced with `yugabyted`.

vdamle-yb committed:
Apply suggestions from code review

Co-authored-by: Dwight Hodge <79169168+ddhodge@users.noreply.github.com>

vdamle-yb committed:
Use yugabyted in example instead of yb-ctl

ddhodge committed:
edit

ddhodge commented on docs/content/preview/secure/tls-encryption/tls-authentication.md:
````diff
@@ -152,6 +152,24 @@ Type "help" for help.
+For example, to create a database, execute the following command:
+
+```sh
+$ ./bin/yugabyted destroy && ./bin/yugabyted create \
````
It's not a 1:1 mapping, yugabyted uses the secure flag instead. Page needs refactoring but I'd say that is for another PR. There is a ticket for it DOC-800

vdamle-yb replied:
The following works for me:
```sh
$ ./bin/yugabyted stop && ./bin/yugabyted start --tserver_flags="ysql_enable_auth=true,$ENABLE_TLS"
```
where 
```sh
$ echo $ENABLE_TLS
use_client_to_server_encryption=true,certs_for_client_dir=/home/ec2-user/var/generated_certs/172.165.30.52/
```

I removed the enable scram flag for the moment because I was running this on master (the diff isn't committed yet). 
Also, because `./bin/yugabyted cert generate_server_certs` generates the certs in the `~/var/` directory by default, and because `yugabyted destroy` seems to clear the whole ~/var/ directory including the certs, I've used stop instead of destroy. I'll change it in the file as well.

ddhodge committed:
Update docs/content/preview/secure/tls-encryption/tls-authentication.md

vdamle-yb committed:
Propagate changes to stable and v2024.2 branches. Change 'create' to 'start'

vdamle-yb committed:
Change destroy to stop

vdamle-yb committed:
Revert changes the preview docs re additions to SCRAM in conn mgr

vdamle-yb commented:
@ddhodge @aishwarya24 Made two commits - one to change `destroy` to `stop` (after verifying the command works locally) and one to revert (most) of my changes to the preview branch. 

Should have addressed most of the comments now? Only unresolved thing is if there is any other issue with using `stop` over `destroy`. Let me know if I should revert that.

ddhodge committed:
Merge branch 'master' into pr/vdamle-yb/28213

ddhodge committed:
Merge branch 'scram-docs' of https://github.com/vdamle-yb/yugabyte-db-docs into pr/vdamle-yb/28213

ddhodge committed:
Update SCRAM support in Conn Mgr

ddhodge committed:
No SCRAM in preview CM

ddhodge committed:
Update examples to yugabyted

ddhodge committed:
edit

ddhodge committed:
update yugabyted examples

ddhodge committed:
format

ddhodge committed:
Merge branch 'master' into scram-docs

ddhodge committed:
comment out changes for 2025.1.1
